embassy-sync/src/signal.rs

```diff
@@ -111,6 +111,17 @@ where
         poll_fn(move |cx| self.poll_wait(cx))
     }
 
+    /// non-blocking method to try and take the signal value.
+    pub fn try_take(&self) -> Option<T> {
+        self.state.lock(|cell| {
+            let state = cell.replace(State::None);
```

Comment:
This will remove the `Waiting` state, so if there was another task waiting with async `wait()` it'll hang.

Reply:
Good catch